core/trino-main/src/main/java/io/trino/operator/output/PagePartitioner.java

```diff
@@ -215,6 +216,7 @@ public void prepareForRelease(OperatorContext operatorContext)
         outputSizeInBytes = adjustFlushedOutputSizeWithEagerlyReportedBytes(outputSizeInBytes);
         bufferedSizeInBytes = adjustEagerlyReportedBytesWithBufferedBytesOnRelease(bufferedSizeInBytes);
         operatorContext.recordOutput(outputSizeInBytes + bufferedSizeInBytes, 0 /* no new positions */);
+        return serializer.getAndResetMetrics();
```

Comment:
nit: you could reset the metrics in `io.trino.operator.output.PagePartitionerPool#release` instead of here, avoiding changing this method

Reply:
I could - but then the internal state of PagePartitioner is off between `PP.prepareForRelease` and `PPP.release`. I prefer the way it is done now.